### PR TITLE
Attack results take goal function result

### PR DESCRIPTION
**Pull request by jxmorris12**

### CODE REVIEW

File: textattack/shared/utils.py [hunk 1]
```diff
@@ -109,6 +109,13 @@ def default_class_repr(self):
         extra_str = ''
     extra_str = extra_str.format(**self.__dict__)
     return f'{self.__class__.__name__}{extra_str}'
+
+def color_text(label, color=None, method=None):
```
Comment (uvafan) [inline textattack/shared/utils.py:113]: Why did you rename this to color_text? Maybe color_label isn't the best name if we're going task-agnostic but I feel like having a color_text and a color_text_by_method for which the color_text changes some stuff then calls color_text_by_method is confusing.

File: textattack/attack_results/attack_result.py [hunk 2]
```diff
@@ -1,85 +1,98 @@
...
-    def __data__(self, color_method=None):
-        data = [self.result_str(color_method=color_method), 
-                self.original_text.text,
-                self.perturbed_text.text]
+    def str_lines(self, color_method=None):
+        """ A list of the lines to be printed for this result's string
+            representation. """
+        lines = [
+            self.goal_function_result_str(color_method=color_method), 
+            self.original_text(),
```
[the remark this answers is not in the document]
Comment (jxmorris12) [reply textattack/attack_results/attack_result.py:49]: thanks, this is nicer

File: textattack/attack_results/failed_attack_result.py [hunk 3]
```diff
@@ -1,16 +1,15 @@
-from textattack.attack_results import AttackResult
+from .attack_result import AttackResult
```
[the remark this answers is not in the document]
Comment (jxmorris12) [reply textattack/attack_results/failed_attack_result.py:1]: I think relative imports are better where possible

File: textattack/attack_results/attack_result.py [hunk 4]
```diff
@@ -1,85 +1,98 @@
...
-        t2 = self.original_text.replace_words_at_indices(new_is, new_w2s)
+        t1 = self.original_result.tokenized_text.replace_words_at_indices(replaced_word_indices, 
+            new_words_1)
+        t2 = self.perturbed_result.tokenized_text.replace_words_at_indices(replaced_word_indices, 
+            new_words_2)
                 
         return t1.clean_text(), t2.clean_text()
+        
+    
+class SuccessfulAttackResult(AttackResult):
```
[the remark this answers is not in the document]
Comment (jxmorris12) [reply textattack/attack_results/attack_result.py:97]: yeah that was the idea.

File: textattack/shared/utils.py (hunk 1 above)
Comment (jxmorris12) [reply textattack/shared/utils.py:113]: thanks! I consolidated these into a single method, `color_text`
